calc/rule_convert.go

```diff
@@ -129,7 +130,31 @@ func parsedRuleToProtoRule(in *ParsedRule) *proto.Rule {
 	}
 
 	if in.HTTPMatch != nil {
-		out.HttpMatch = &proto.HTTPMatch{Methods: in.HTTPMatch.Methods}
+		out.HttpMatch = &proto.HTTPMatch{}
+		paths := []*proto.HTTPMatchPathMatchTypes{}
```

Comment:
Prefer `var paths []*proto.HTTPMatchPathTypes` over this.  The difference is that it avoids allocating a 0-length slice if it's not needed (some of the linters flag this up).

```
len(nil) == 0
```
so it's safe to do that.

Reply:
good to know. I initially had it as a var.
Will add a few fv's as well. Thanks for the review.